handlers/http_test.go

```diff
@@ -226,8 +226,68 @@ func TestHandleWhenErrWithoutStatus(t *testing.T) {
 	limiter.ServeHTTP(w, r)
 	assertNoRateLimitHeaders(t, w.Header())
 
-	if w.Code != 500 {
-		t.Fatalf("expected status 500, received %d", w.Code)
+	if w.Code != http.StatusInternalServerError {
+		t.Fatalf("expected status %d, received %d", http.StatusInternalServerError, w.Code)
 	}
 	limitMock.AssertExpectations(t)
 }
+
+// Test cases when an error occurs and either value of AllowOnError
+var allowOnErrorCases = []struct {
+	AllowOnError      bool
+	ExpectedCode      int
+	RateLimiterStatus []ratelimiter.Status
+}{
+	// It should still block if AllowOnError == false
+	{
+		false,
+		http.StatusInternalServerError,
+		[]ratelimiter.Status{},
+	},
+	// If AllowOnError == true and
+	{
+		true,
+		http.StatusOK,
+		[]ratelimiter.Status{},
+	},
+	// It should still add headers if AllowOnError == true
+	{
+		true,
+		http.StatusOK,
+		[]ratelimiter.Status{sphinxStatus},
+	},
```

Comment:
when i was reading this, it seemed like these last two cases were the same, because i didnt realize that the third field (`RateLimiterStatus`) was part of the input, not the expected results. maybe reordering the fields in the test spec would make that clearer? or adding an explanatory comment